store/tikv/region_cache.go

```diff
@@ -108,10 +111,32 @@ func (r *RegionStore) follower(seed uint32) int32 {
 			continue
 		}
 		if r.storeFails[followerIdx] == atomic.LoadUint32(&r.stores[followerIdx].fail) {
-			return followerIdx
+			if len(labelKeys) == 0 {
+				return followerIdx
+			}
+			candidates = append(candidates, followerIdx)
 		}
 		seed++
 	}
+	bestMatchedLevel := -1
+	var bestCandidate int32 = -1
+	for _, idx := range candidates {
+		matchedLevel := 0
+		for i, label := range r.stores[idx].labels {
+			if i < len(labelKeys) {
+				if label.Key == labelKeys[i] && label.Value == labelValues[i] {
+					matchedLevel += 1
+				}
+			}
+		}
+		if matchedLevel > bestMatchedLevel {
+			bestMatchedLevel = matchedLevel
+			bestCandidate = idx
+		}
+	}
+	if bestCandidate >= 0 {
+		return bestCandidate
+	}
```

Comment:
maybe also take care this `else`, we should better try other follower instead of using leader directly if all stores in label set have down but still some other alive follower store can try this request.

Reply:
Yes, it's already considered. You can see the initial `bestMatchedLevel` is -1, so if all stores with matched labels fail, the first active follower will be choosen.

Reply:
great, but this is complex, how about change those to "an sort question" that make label marched on front..

using sort way, we can add more complex logic in simple way, e.g. we can sort un-label nodes with latency..

at last, maybe follower read can using leader if leader is more near to callee then other followers.

Reply:
or maybe I can try sort way in next PR :smile: